pkg/provider/aws/knowledgebase/iam.go

```diff
@@ -18,9 +18,17 @@ var IamKB = knowledgebase.Build(
 			return nil
 		},
 	},
+	knowledgebase.EdgeBuilder[*resources.IamRole, *resources.Secret]{
+		Configure: func(role *resources.IamRole, secret *resources.Secret, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			secretPolicyDoc := resources.CreateAllowPolicyDocument([]string{"secretsmanager:DescribeSecret", "secretsmanager:GetSecretValue"}, []core.IaCValue{{Resource: secret, Property: resources.ARN_IAC_VALUE}})
+			inlinePol := resources.NewIamInlinePolicy(fmt.Sprintf("%s-secretpolicy", secret.Name), core.DedupeAnnotationKeys(append(role.ConstructsRef, secret.ConstructsRef...)), secretPolicyDoc)
+			role.InlinePolicies = append(role.InlinePolicies, inlinePol)
+			return nil
+		},
+	},
 	knowledgebase.EdgeBuilder[*resources.IamPolicy, *resources.SecretVersion]{
 		Configure: func(policy *resources.IamPolicy, secretVersion *resources.SecretVersion, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
-			secretPolicyDoc := resources.CreateAllowPolicyDocument([]string{"secretsmanager:GetSecretValue"}, []core.IaCValue{{Resource: secretVersion.Secret, Property: resources.ARN_IAC_VALUE}})
+			secretPolicyDoc := resources.CreateAllowPolicyDocument([]string{"secretsmanager:DescribeSecret", "secretsmanager:GetSecretValue"}, []core.IaCValue{{Resource: secretVersion.Secret, Property: resources.ARN_IAC_VALUE}})
```

Comment:
do we need this policy on both of these here? I know that the rds dep i believe is the secretversion, so that wouldnt need describe. Im assuming that config vs secrets uses inline vs policy? Describe doesnt hurt, but a thought for the future is how do we know read vs write when we want to have granular permissions

Reply:
I'm not sure we should be mapping anything directly to the SecretVersion. If the user updated the secret, that would invalidate all the policies to read it. The permissions on the Secret should suffice for everything we need to do.

Reply:
@gordon-klotho I believe you're right. That edge was created as part of the RDS expansion change, but I can change it to `Secret` and verify that it still works.

Reply:
@jhsinger-klotho yeah, dealing with more granular permissions is something we'll need to deal with at some point.